### ISSUE TITLE
As people are added, the site is becoming sluggish and slow to load

### ISSUE THREAD

**Issue by sam-cross** [wesbos/awesome-uses#579]
Not sure what could be done about this but I've just noticed it. With the amount of people who are in the list now, the website loads up quite slowly.

Could lazy-loading be implemented somehow?

**Comment by wesbos** [comment]
How is it slow? Scrolling? Images loading? rendering?

**Comment by sam-cross** [comment]
Using Firefox, these are the network stats I got:

```
262 requests
3.86 MB / 1.99 MB transferred
Finish: 11.96 s
DOMContentLoaded: 4.97 s
load: 11.53 s
```

A lot of the request were avatar URLs - these could possibly be minimised by only loading when they're in the viewport.

Filtering is also extremely slow for me but that could be due to my system (dual core i3).

**Comment by wesbos** [comment]
Yeah almost all of that is image requests. I'll add lazy loading, looks like it will be added to firefox soon so its a good solution. 

Filtering does have a few MS of lag for me too. Likely in the React render. Anyone want to take a look at this?

**Comment by wesbos** [comment]
https://github.com/wesbos/awesome-uses/commit/ee7be1dcdf2f436da4bc34a51e78a4b7bd31ebf7

**Comment by wesbos** [comment]
You can tell the lag is in the render because when we filter for small numbers, it's fast. 

I bet we can just hide people instead of re-rendering each time. Let me try that..

**Comment by wesbos** [comment]
I goofed around with it for a bit yesterday but I don't think hiding it is the issue.. 

If anyone is good ith the profiler on react dev tools, please take a look. It very clearly shows that the render for the large tags( 150+ people) takes ~300ms which is way too long

**Comment by iamandrewluca** [comment]
Notes:
* https://reactjs.org/docs/optimizing-performance.html#virtualize-long-lists

**Comment by sam-cross** [comment]
[This](https://github.com/bvaughn/react-virtualized/blob/master/source/AutoSizer/AutoSizer.example.js) looks incredibly promising, and I don't see it being too difficult to implement. Seems like it'll just take a few lines. I'll play around.

**Comment by iamandrewluca** [comment]
Or maybe infinite pagination can be added.
https://www.gatsbyjs.org/docs/adding-pagination/

**Comment by jonaskuske** [comment]
I think the easiest (potential) fix would be to apply `content-visibility` and `contain-intrinsic-size` in CSS, so only the visible parts are rendered by the browser: https://web.dev/content-visibility/
